File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        res = ""
        Open=0
        close=0
        start = 0
        for i in range(len(s)):
            if s[i]=='(':
                Open+=1
            elif s[i]==')':
                close+=1
            
            if Open==close:
                res+=s[start+1:i]
                start = i+1
                Open = 0
                close = 0
          
        return res
                
        

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string result = "";
        int i=0;
        int n = s.length();
        int open = 0;
        int close = 0;
        int start = 0;
        while(i<n){
            if(s[i]=='(') open++;
            else if(s[i]==')') close++;
            
            if(open==close){
                string sub = s.substr(start+1,open+close-2);
                open=0;
                close=0;
                start = i+1;
                result += sub;
            }
            
            i++;
        }
        return result;
    }
};